lib/grape/dsl/settings.rb

```diff
@@ -162,6 +162,18 @@ def within_namespace(&_block)
 
         result
       end
+
+      private
+
+      # Obtains the current class :inheritable_setting. If available, it returns the superclass's :inheritable_setting.
+      # Otherwise, a clean :inheritable_setting is returned.
+      def obtain_top_level_setting
```

Comment:
Agreed! Updated in https://github.com/ruby-grape/grape/pull/1665/commits/1a3ba3f0cb47a2a3f20bffa682c03718ab6c2694